src/games/planetwars/visualizer/visualsHelper.js

```diff
@@ -0,0 +1,477 @@
+    var sy = Math.cos(w) + t1 * Math.sin(w);
+
+    var degrees = space_math.toDegrees(Math.atan2(sy, sx));
+    return 'rotate(' + (degrees + 180) % 360 + ')';
+  }
+
+  static homannAngle(turn, distance) {
+    if (!distance) distance = space_math.euclideanDistance(this.origin, this.destination);
+    var mod = turn / distance;
+    return mod * (Math.PI * 2) - Math.PI;
+  }
+}
+
+class Scores {
+  static addScores(d3selector, color_map, scores) {
+    var start_y = 50;
+    var size = 30;
+    Scores.max_bar_size = 100;
+
+    d3selector.attr("font-family", "sans-serif")
+      .attr("font-size", 14 + "px")
+      .attr('fill', d => color_map[d.player]);
+    d3selector.append('circle')
+      .attr('r', d => 5)
+      .attr('cx', d => "5%")
+      .attr('cy', (d, i) => start_y + size * i);
+    d3selector.append('text')
+      .attr('class', 'player_name')
+      .attr('x', d => "15%")
+      .attr('y', (d, i) => start_y + 5 + size * i)
+      .text(d => d.player);
+    d3selector.append('text')
+      .attr('class', 'planet_count')
+      .attr('x', d => "45%")
+      .attr('y', (d, i) => start_y + 5 + size * i)
+      .text(d => d.planets);
+    d3selector.append('circle')
+      .attr('r', d => "3%")
+      .attr('cx', d => "55%")
+      .attr('cy', (d, i) => start_y - 1 + size * i)
+      .attr('fill', 'url(#earth)')
+      .attr('stroke', d => color_map[d.player]);
+    var end_y = 0;
+    d3selector.append('text').attr('class', 'strength')
+      .attr('x', d => "80%")
+      .attr('y', (d, i) => {
+        end_y = start_y + 5 + size * i;
+        return end_y;
+      })
+      .text((d, i) => d.strengths[i] + " \u2694");
+    end_y += 20;
+
+    d3selector.append('rect').attr('class', 'ratioblock')
+      .attr('x', (d, i) => {
+        var strength_before = 0;
+        if (i != 0) {
+          for (var j = 0; j < i; j++) {
+            strength_before += d.strengths[j];
+          }
+        }
+        return Scores.max_bar_size * (strength_before / d.total_strength) + '%';
+      })
+      .attr('y', (d, i) => end_y + 20)
+      .attr('width', (d, i) => (Scores.max_bar_size * (d.strengths[i] / d.total_strength)) + '%')
+      .attr('height', 20)
+      .append('title').text(d => Helper.visualOwnerName(d.player));
+
+  }
+
+  static update(d3selector) {
+    d3selector.select('.planet_count').text(d => d.planets);
+    d3selector.select('.strength').text((d, i) => d.strengths[i] + " \u2694");
+    d3selector.select('.ratioblock').attr('x', (d, i) => {
+        var strength_before = 0;
+        if (i != 0) {
+          for (var j = 0; j < i; j++) {
+            strength_before += d.strengths[j];
+          }
+        }
+        return Scores.max_bar_size * (strength_before / d.total_strength) + '%';
+      })
+      .attr('width', (d, i) => (Scores.max_bar_size * (d.strengths[i] / d.total_strength)) + '%');
+  }
+}
+
+// TODO: this modefies the model, this should perhaps generate a new model that is vis only
+class Preprocessor {
```

Comment:
I feel like a the preprocessor might need a seperate file, certainly since it's function will change significantly when we update to unfinished data/streaming. Also, 500 lines is quite a lot for a file.

Reply:
Agree, I would also maybe want to split off expeditions but then that's inconsistent compared to the other visual objects. What do you think?

Reply:
If you split of Expeditions, you need to split off all of it. Which might be not too bad of an idea. In a separate visuals directory? Might be confusing that for the data, they are grouped, but for the visuals they are split.